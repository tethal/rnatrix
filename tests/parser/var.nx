var a = 21;
a = a * 2;
