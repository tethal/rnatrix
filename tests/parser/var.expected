Program @<string>:5:1-1
  FunDecl("main") @<string>:1:5-9
    Block @<string>:1-12:4-2
      VarDecl @<string>:2:5-16
        name: "a" @<string>:2:9-10
        IntLiteral(21) @<string>:2:13-15
      Assign @<string>:3:5-14
        Var("a") @<string>:3:5-6
        Binary @<string>:3:9-14
          op: Mul @<string>:3:11-12
          Var("a") @<string>:3:9-10
          IntLiteral(2) @<string>:3:13-14
